src/swarm-language/Swarm/Language/Syntax.hs

```diff
@@ -145,6 +145,9 @@ data RelativeDir = DLeft | DRight | DBack | DForward | DDown
 data Direction = DAbsolute AbsoluteDir | DRelative RelativeDir
   deriving (Eq, Ord, Show, Read, Generic, Data, Hashable, ToJSON, FromJSON)
 
+-- | TODO: unify Heading without backward dependency?
+type Heading = V2 Int32
```

Comment:
@byorgey do you have any suggestions on how to resolve this?

Maybe we could split off a `Swarm.Language.Direction` which could export `Heading`. :thinking:

Reply:
I think we should just move all this DirInfo and heading stuff to `Swarm.Game.Direction` or something like that.  It is all about the meaning/semantics of aspects of the language, which really doesn't belong here.

Reply:
To expand on my previous comment, which was written hastily on a phone: stuff in `Swarm.Language.*` is about defining the *static* aspects of the language, e.g. the syntax and type system.  Stuff in `Swarm.Game.*` defines the *dynamic* aspects of the language, i.e. the *meaning* of things in the language and what they do/are at runtime (plus a lot of other game stuff that is inextricably tied to the language semantics).  So the `Direction` type definitely belongs here in `Swarm.Language.Syntax`. But all the stuff about converting `Direction`s to `Heading`s really should go in `Swarm.Game.*` somewhere.

Reply:
Done in #1103. :+1: